src/rgw/rgw_rest_s3.cc

```diff
@@ -5870,3 +5874,201 @@ bool rgw::auth::s3::S3AnonymousEngine::is_applicable(
 
   return route == AwsRoute::QUERY_STRING && version == AwsVersion::UNKNOWN;
 }
+
+
+const char *RGWS3Select::header_name_str[3] = {":event-type", ":content-type", ":message-type"};
+const char *RGWS3Select::header_value_str[3] = {"Records", "application/octet-stream", "event"};
+#include "s3select.h"
+
+RGWS3Select::RGWS3Select()
+{
+  set_get_data(true);
```

Comment:
[nit] use initialization line for data members (also, make them const if possible)

Reply:
i used existing API (did not create it)

Reply:
i didn't put the comment on the right line, anyway, what i meant is:
```
RGWS3Select::RGWS3Select() : 
chunk_number(0), 
s3select_syntax(new s3select()), 
m_processed_bytes(0), 
m_previous_line(false) {
  set_get_data(true);
}
```

Reply:
ok

Reply:
yes, use ctor-initializer whenever possible